packages/eslint-plugin/docs/rules/unified-signatures.md

```diff
@@ -0,0 +1,31 @@
+# Warns for any two overloads that could be unified into one by using a union or an optional/rest parameter. (unified-signatures)
+
+Warns for any two overloads that could be unified into one by using a union or an optional/rest parameter.
+
+## Rule Details
+
+This rule aims to keep the source code as maintanable as posible by reducing the amount of overloads.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+function f(x: number): void;
+function f(x: string): void;
+```
+```ts
+f(): void;
+f(...x: number[]): void;
+```
+
+Examples of **correct** code for this rule:
+
+```ts
+function f(x: number | stting): void;
```

Comment:
```suggestion
function f(x: number | string): void;
```

Reply:
![image](https://user-images.githubusercontent.com/900523/52124159-5634e380-25f6-11e9-9a94-e908c0271c12.png)

Reply:
```ts
interface Sting extends EnglishmanInNewYork {}
```

Reply:
Ok, I'll stop...

Reply:
Haha good one, thanks for the laugh. Fixed.